otaclient/app/boot_control/_grub.py

```diff
@@ -578,201 +609,99 @@ def _ensure_ota_partition_symlinks(self):
+                force_initialize=self._boot_control.initialized,
             )
-
-            # refroot mount point
-            self.ref_slot_mount_point = Path(cfg.ACTIVE_ROOT_MOUNT_POINT)
-            # try to umount refroot mount point
-            CMDHelperFuncs.umount(self.ref_slot_mount_point)
-            if not os.path.isdir(self.ref_slot_mount_point):
-                os.mkdir(self.ref_slot_mount_point)
-
-            # init boot control
-            #   1. load/process ota_status
-            #   2. finalize update/rollback or init boot files
-            self._init_boot_control()
         except Exception as e:
             logger.error(f"failed on init boot controller: {e!r}")
             raise BootControlInitError from e
 
-    def _init_boot_control(self):
-        # load ota_status str and slot_in_use
-        _ota_status = self._load_current_ota_status()
-        _slot_in_use = self._load_current_slot_in_use()
-
-        # NOTE: for backward compatibility, only check otastatus file
-        if not _ota_status:
-            logger.info("initializing boot control files...")
-            _ota_status = wrapper.StatusOta.INITIALIZED
-            self._boot_control.init_active_ota_partition_file()
-            self._store_current_slot_in_use(self._boot_control.active_slot)
-            self._store_current_ota_status(wrapper.StatusOta.INITIALIZED)
-
-        # populate slot_in_use file if it doesn't exist
-        if not _slot_in_use:
-            self._store_current_slot_in_use(self._boot_control.active_slot)
-
-        if _ota_status in [wrapper.StatusOta.UPDATING, wrapper.StatusOta.ROLLBACKING]:
-            if self._is_switching_boot():
-                self._boot_control.finalize_update_switch_boot(
-                    abort_on_standby_missed=True
-                )
-                # switch ota_status
-                _ota_status = wrapper.StatusOta.SUCCESS
-            else:
-                if _ota_status == wrapper.StatusOta.ROLLBACKING:
-                    _ota_status = wrapper.StatusOta.ROLLBACK_FAILURE
-                else:
-                    _ota_status = wrapper.StatusOta.FAILURE
-        # other ota_status will remain the same
-
-        # detect failed reboot, but only print error logging
-        if (
-            _ota_status != wrapper.StatusOta.INITIALIZED
-            and _slot_in_use
-            and _slot_in_use != self._boot_control.active_slot
-        ):
-            logger.error(
-                f"boot into old slot {self._boot_control.active_slot}, "
-                f"but slot_in_use indicates it should boot into {_slot_in_use}, "
-                "this might indicate a failed finalization at first reboot after update/rollback"
-            )
-
-        # apply ota_status to otaclient
-        self.ota_status = _ota_status
-        self._store_current_ota_status(_ota_status)
-        logger.info(f"boot control init finished, ota_status is {_ota_status}")
-
-    def _is_switching_boot(self):
```

Comment:
In this PR, boot_control._common._is_switching_boot is used?
(If so, updating from old otaclient to this otaclient doesn't work properly)

Reply:
The backward compatibility is ensured when loading `slot_in_use` file:
https://github.com/tier4/ota-client/blob/9b165fd6137937346cc04d693604f6f10adaa63a/otaclient/app/boot_control/_common.py#L525-L544

If the `slot_in_use` file is not presented, `slot_in_use` file will be init and set to current booted slot.  
`_load_slot_in_use_file` is called before parsing ota_status and detecting switch boot.